### ISSUE TITLE
Pandoc candidate generator

### ISSUE THREAD

**Issue by zkry** [zkry/p-search#16]
There should be a new candidate generator that works like the FILESYSTEM one, except it searches for a specific file format, using pandoc, converts them to text and stores them in a temporary directory, then uses tools like rg to search on them.

**Comment by swflint** [comment]
It may make sense to build a dispatch mechanism of some sort, to allow for even more file types to be supported.  For example, I don't believe pandoc can convert PDF to markdown, but there are tools that can.

**Comment by zkry** [comment]
Good idea! I saw this project a few weeks ago: https://github.com/microsoft/markitdown which seems to be a tool along those lines. Maybe I could either use this or create something like it Elisp as like you said, there are CLI tools for each of these.

**Comment by swflint** [comment]
Personally, I'd prefer something configurable in elisp.  For example, something that lets me configure it using an alist of regexps to functions that are called on the filename, and return a text-only filename/buffer.
